src/dune_rules/coq_rules.ml

```diff
@@ -191,10 +191,13 @@ let select_native_mode ~sctx ~dir (buildable : Coq_stanza.Buildable.t) =
     else if buildable.coq_lang_version < (0, 7) then Memo.return Coq_mode.VoOnly
     else
       let* coqc = resolve_program sctx ~dir ~loc:buildable.loc "coqc" in
-      let+ config = Coq_config.make ~coqc in
-      match Coq_config.by_name config "coq_native_compiler_default" with
-      | Some (`String "yes") | Some (`String "ondemand") -> Coq_mode.Native
-      | _ -> Coq_mode.VoOnly)
+      let+ config = Coq_config.make_opt ~coqc in
+      match config with
+      | None -> Coq_mode.VoOnly
```

Comment:
A case where we might not want this to happen is if we compose with the coq repo with native enabled and `coqc --config` doesn't work properly. Then this will silently not have the native rules. Perhaps a warning should be put here when this happens?